errbot/backends/slack.py

```diff
@@ -650,16 +651,16 @@ def build_identifier(self, txtrep):
         log.debug("building an identifier from %s" % txtrep)
         username, userid, channelname, channelid = self.extract_identifiers_from_string(txtrep)
 
+        if userid is None and username is not None:
+            userid = self.username_to_userid(username)
+        if channelid is None and channelname is not None:
+            channelid = self.channelname_to_channelid(channelname)
+        if userid is not None and channelid is not None:
+            return SlackRoomOccupant(self.sc, userid, channelid, bot=self)
```

Comment:
Indeed there is a bug, my brain was foggy ;)

Still I think there is a bug left (it was there before probably):
`#channel` should not return a SlackRoom ? (ie there is a channelname and no userid -> Room)